packages/proxy/lib/http/util/regex-rewriter.ts

```diff
@@ -17,10 +17,22 @@ const topOrParentLocationOrFramesRe = /([^\da-zA-Z\(\)])?(\btop\b|\bparent\b)([.
 const jiraTopWindowGetterRe = /(!function\s*\((\w{1})\)\s*{\s*return\s*\w{1}\s*(?:={2,})\s*\w{1}\.parent)(\s*}\(\w{1}\))/g
 const jiraTopWindowGetterUnMinifiedRe = /(function\s*\w{1,}\s*\((\w{1})\)\s*{\s*return\s*\w{1}\s*(?:={2,})\s*\w{1}\.parent)(\s*;\s*})/g
 
-const integrityTagReplacementRe = new RegExp(`(${STRIPPED_INTEGRITY_TAG}|integrity)(=(?:\"|\')sha(?:256|384|512)-.*?(?:\"|\'))`, 'g')
+const buildIntegrityReplacementRe = (isHtml = true) => {
+  if (!isHtml) {
+    // only replace integrity if a trailing period (.) or string exists inside JS/Other resources
```

Comment:
Could you add more code comments here to help explain why there is a difference? My head already hurts just trying to parse these regexes.

Reply:
I can give it a shot. I am having issues thinking of something useful though so I'm going to try typing my thoughts out 😆 . The main goal here is only replace integrity in JavaScript if the word `integrity` or `cypress-stripped-integrity` is preceded by a `.` or quotes. So something like 

```
foo.integrity = 'sha256-some-hash'
var integrity = 'sha256-some-hash'
```

becomes
```
foo['cypress-stripped-integrity'] = 'sha256-some-hash'
var integrity = 'sha256-some-hash'

```

It isn't so much what we match on as much as it is what we replace it with (the brackets vs the tagname). At some point in the future I am starting to think we might want to apply both of these because JS like above could be in HTML, but I am a bit hesitant to try it because it could have side effects we aren't testing for, and we haven't run into that problem yet. Now that I think about it I might want to go. Did any of that make sense or is it more of a jumbled ramble?

Reply:
I gave it ago and it's complicated with some added lookbacks 😄. I think we might want to hold off on that for now  until it actually surfaces as a problem (if it even is one). Maybe we can meet up on Monday to see if we can improve the documentation?

Reply:
I actually got it to work! I tried adding some docs in 41e82c835d86888f5aae2c71cc1d81f5b9977252 but I don't think they are really great. It also means the regex is more complicated, but handles more cases correctly without sacrificing performance. Regardless of resource type, both regex functions need to execute because HTML can have nested javascript inside of it, and both regexes need to run to rewrite correctly